### PR TITLE
dashboard: Use go-bindata

### PR DESCRIPTION
**Pull request by jvatic**

### CODE REVIEW

**Comment by titanous** [comment]
Vendoring should be done in a separate commit.

**Comment by jvatic** [comment]
@titanous ready for another view pass

**Comment by titanous** [comment]
Commit message needs WIP removed.

File: util/rubyassetbuilder/build.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    --workdir /build \
+    --user $(id -u) \
+    flynn/${target}-builder \
+    bash -c "cp --recursive /app/.bundle . && cp --recursive /app/vendor/bundle vendor/ && bundle exec rake compile"
+
+  mkdir app
+  mv build app
+  cd ${ROOT}/${target}
+  ./bin/go-bindata -prefix ${tmpdir}/ -pkg main ${tmpdir}/app/build/...
+  gofmt -s -w bindata.go
```
Comment (titanous) [inline util/rubyassetbuilder/build.sh]: This gofmt should be unecessary?

**Comment by titanous** [comment]
The Dockerfile refers to `/app/Gemfile`, how does that work?

**Comment by titanous** [comment]
Ah, just found the `cp`.

File: dashboard/api.go [hunk 2]
```diff
@@ -176,19 +208,13 @@ type DashboardConfig struct {
 }
 
 func serveDashboardJs(res http.ResponseWriter, req *http.Request, conf *Config) {
-	file := filepath.Join(conf.StaticPath, "assets", filepath.Base(req.URL.Path))
-	f, err := os.Open(file)
+	path := filepath.Join(conf.StaticPath, "assets", filepath.Base(req.URL.Path))
```
Comment (titanous) [inline dashboard/api.go]: Should `StaticPath` be removed?

File: dashboard/api.go [hunk 3]
```diff
@@ -200,7 +226,7 @@ func serveDashboardJs(res http.ResponseWriter, req *http.Request, conf *Config)
 	})
 	jsConf.Write([]byte(";\n"))
 
-	r := ioutil.NewMultiReadSeeker(bytes.NewReader(jsConf.Bytes()), f)
+	r := ioutil.NewMultiReadSeeker(bytes.NewReader(jsConf.Bytes()), data)
 
-	http.ServeContent(res, req, file, fi.ModTime(), r)
+	http.ServeContent(res, req, path, time.Now(), r)
```
Comment (titanous) [inline dashboard/api.go]: Does using `time.Now` affect caching?

File: dashboard/api.go (hunk 2 above)
Comment (titanous) [inline dashboard/api.go]: Might make sense to cache the mtime of the binary at start.

File: dashboard/api.go (hunk 3 above)
Comment (titanous) [inline dashboard/api.go]: Might make sense to cache the mtime of the binary at start.

File: dashboard/api.go (hunk 2 above)
Comment (titanous) [inline dashboard/api.go]: Ah, I guess it's useful in development mode?

File: util/rubyassetbuilder/build.sh [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    --volume "${tmpdir}:/build" \
+    --workdir /build \
+    --user $(id -u) \
+    flynn/${target}-builder \
+    bash -c "cp --recursive /app/.bundle . && cp --recursive /app/vendor/bundle vendor/ && bundle exec rake compile"
+
+  mkdir app
+  mv build app
+  cd ${ROOT}/${target}
+  ./bin/go-bindata -prefix ${tmpdir}/ -pkg main ${tmpdir}/app/build/...
```
Comment (titanous) [inline util/rubyassetbuilder/build.sh]: Add `-nomemcopy`

**Comment by titanous** [comment]
Would https://github.com/elazarl/go-bindata-assetfs simplify anything?

File: dashboard/api.go (hunk 3 above)
Comment (jvatic) [inline dashboard/api.go]: Looks like there's `AssetInfo()` which includes a timestamp, I'll use that.

File: dashboard/api.go (hunk 2 above)
Comment (jvatic) [inline dashboard/api.go]: Yeah, the reason it was added is now gone.

**Comment by jvatic** [comment]
No, go-bindata-assetfs has some issues.

**Comment by jvatic** [comment]
@titanous ready for another pass

File: dashboard/api.go [hunk 5]
```diff
@@ -22,15 +23,24 @@ type LoginInfo struct {
 	Token string `json:"token" form:"token"`
 }
 
+func AssetReader(path string) (io.ReadSeeker, time.Time, error) {
+	data, err := Asset(path)
+	if err != nil {
+		return nil, time.Now(), err
```
Comment (titanous) [inline dashboard/api.go]: Return a `time.Time{}`

**Comment by titanous** [comment]
LGTM pending last comment.

**Pull request merged**